nautobot/extras/tests/test_jobs.py

```diff
@@ -222,10 +235,14 @@ def test_ip_address_vars(self):
             data = job_class.serialize_data(form.cleaned_data)
 
             # Run the job and extract the job payload data
-            run_job(data=data, request=None, commit=False, job_result_pk=job_result.pk)
+            run_job(data=data, request=self.request, commit=True, job_result_pk=job_result.pk)
```

Comment:
Still concerned about this one, sorry. `commit=False` is a user-specified input allowing a job to be run as a "dry run". We absolutely need to still be able to record and view logs generated from such a run, otherwise the dry run is useless since the logs will not be inspectable.

From what you say I gather that maybe this is only an issue with the test automation since we don't have the actual `job_logs` secondary DB connection - is that the case here, that dry run works fine in actual practice but we're just losing the ability to include automated tests for it?

Also, I don't think you answered my first question - are we no longer able to call `run_job(request=None, ...)`? That's needed for at least the case of `nautobot-server run_job` without specifying a `--user` to run the job as.

Reply:
Nautobot_server run_job I think uses the Admin user, but I will check. I had to add this in order to get the job to run correctly.

Reply:
> Also, I don't think you answered my first question - are we no longer able to call `run_job(request=None, ...)`? That's needed for at least the case of `nautobot-server run_job` without specifying a `--user` to run the job as.

We are absolutely able to still use request=None. I will add a test for this. The problem I encountered is that I need to commit in order to test for the logs. And in order to commit, you require a request object due to webhooks.

Reply:
> Still concerned about this one, sorry. `commit=False` is a user-specified input allowing a job to be run as a "dry run". We absolutely need to still be able to record and view logs generated from such a run, otherwise the dry run is useless since the logs will not be inspectable.
> 
> 
> 
> From what you say I gather that maybe this is only an issue with the test automation since we don't have the actual `job_logs` secondary DB connection - is that the case here, that dry run works fine in actual practice but we're just losing the ability to include automated tests for it?


That is correct, we lose the ability for tests to store logs outside of a commit due to the database rollback that happens when we don't commit changes. In order to support this in tests, we would need to make JobLogEntries and at least JobResults be their own real database. It has to be real not fake like we are doing here in order for Django test to do proper database cleanup. 

When using separate real databases, you can lose relationship functionality between  tables that exist on the two separate databases. This would definitely be a breaking change and something we can look to refactor in 2.0 if that is alright.

Reply:
To add some more fun to this, **IF** we did want to go down the path of moving JobLogEntry and JobResult to their own real database, we would need to remove the ForeignKey on JobResult.obj_type due to https://docs.djangoproject.com/en/3.1/topics/db/multi-db/#cross-database-relations. This would likely pose issues though, so would need further discussion.

Reply:
@glennmatthews There are already tests that use `run_job(request=None, commit=False...)` such as the test_ready_only_job_pass, I just didn't modify that test in the PR, so it doesn't show up in the diffs... There are also tests of the management command that do the same. So I believe we are covered for this.

>>Also, I don't think you answered my first question - are we no longer able to call run_job(request=None, ...)? That's needed for at least the case of nautobot-server run_job without specifying a --user to run the job as.

>We are absolutely able to still use request=None. I will add a test for this.